### DISCUSSION TITLE
Idea: Generate a Quarkiverse BOM with all the tested and released Quarkiverse extensions using the Quarkus Core version as a classifier

### DISCUSSION THREAD

**Discussion by gastaldi** [quarkusio/quarkus#21918] [Community]
I've been thinking about the [Quarkiverse extensions](https://github.com/search?q=topic%3Aquarkus-extension+org%3Aquarkiverse+fork%3Atrue) and how would someone know which version to use on a specific Quarkus core version. Although our tooling has the knowledge to include that in new projects, it would be nice if one could use a BOM containing all the quarkiverse extensions compatible with a specific Quarkus core version, so one could use like the following: 

Here is an example in the pom.xml. Instead of: 

```xml
<properties>
  <quarkus.version>2.5.1.Final</quarkus.version>
  <quarkus-hibernate-types.version>0.2.0</quarkus-hibernate-types.version>
</properties>
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>io.quarkus.platform</groupId>
      <artifactId>quarkus-bom</artifactId>
      <version>${quarkus.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
<dependencies>
  <dependency>
    <groupId>io.quarkiverse.hibernatetypes</groupId>
    <artifactId>quarkus-hibernate-types</artifactId>
    <version>${quarkus-hibernate-types.version}</version>
  </dependency>
  <dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy-jackson</artifactId>
  </dependency>
</dependencies>
```

We'd have something like: 

```xml
<properties>
  <quarkus.version>2.5.1.Final</quarkus.version>
  <quarkiverse.version>1</quarkiverse.version>
</properties>
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>io.quarkus.platform</groupId>
      <artifactId>quarkus-bom</artifactId>
      <version>${quarkus.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
    <dependency>
      <groupId>io.quarkiverse</groupId>
      <artifactId>quarkiverse-bom</artifactId>
      <version>${quarkiverse.version}</version>
      <classifier>${quarkus.version}</classifier>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
<dependencies>
  <dependency>
    <groupId>io.quarkiverse.hibernatetypes</groupId>
    <artifactId>quarkus-hibernate-types</artifactId>
  </dependency>
  <dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy-jackson</artifactId>
  </dependency>
</dependencies>
```

The `io.quarkiverse:quarkiverse-bom` would be auto-generated and deployed to maven central automatically based on the outcome of the extension tests triggered when a new Quarkus version is out.

**Comment by aloubyansky** [comment]
That means the list of extensions will be changing from one version of the `quarkiverse-bom` to another. E.g. if an extension failed the test, it wouldn't be included. But once it's fixed, it would re-appear. Users won't know whether an extension in their projects is included in a given BOM version or not.

**Reply by gastaldi** [reply]
In this case if an extension fails a test, the bom would keep the original version and will not be updated in the bom

**Comment by maxandersen** [comment]
is this not better as a automatic generated platform ? as in there isn't much fun getting list of extensiosn that worked against core if the transitive deps of these are not all alighed...but we could check that with a generated platform.
